pkg/apis/crd/v1alpha2/types.go

```diff
@@ -205,6 +205,23 @@ type EgressStatus struct {
 	// EgressIP indicates the effective Egress IP for the selected workloads. It could be empty if the Egress IP in spec
 	// is not assigned to any Node. It's also useful when there are more than one Egress IP specified in spec.
 	EgressIP string `json:"egressIP"`
+
+	Conditions []EgressCondition `json:"conditions,omitempty"`
+}
+
+type EgressConditionType string
+
+const (
```

Comment:
We just introduced Egress API's v1beta1 version, please rebase on main branch and add the conditions to both versions.